eng/mgmt/changelog/src/main/java/com/azure/resourcemanager/tools/changelog/changelog/DefinitionStageChangeLog.java

```diff
@@ -0,0 +1,101 @@
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class DefinitionStageChangeLog extends ChangeLog {
+    private String parentClass;
+    private List<Set<JApiMethod>> oldMethodStages;
+    private List<Set<JApiMethod>> newMethodStages;
+
+    DefinitionStageChangeLog(Map<String, AllMethods> allStages, String parentClass) {
+        this.parentClass = parentClass;
+        oldMethodStages = new ArrayList<>();
+        newMethodStages = new ArrayList<>();
+        AllMethods blankStage = allStages.entrySet().stream().filter(x -> ClassName.name(x.getKey()).equals("Blank")).findAny().get().getValue();
+        calcMethodStages(blankStage, allStages, oldMethodStages, x -> x.getReturnType().getOldReturnType());
+        calcMethodStages(blankStage, allStages, newMethodStages, x -> x.getReturnType().getNewReturnType());
+        calcChangeLog();
+    }
+
+    void calcMethodStages(AllMethods blankStage, Map<String, AllMethods> allStages, List<Set<JApiMethod>> results, Function<JApiMethod, String> getReturnType) {
+        results.add(new HashSet<>(blankStage.getMethods()));
+        Set<JApiMethod> used = new HashSet<>(blankStage.getMethods());
+        for (int i = 0; i < results.size(); ++i) {
+            Set<JApiMethod> nextStage = new HashSet<>();
+            results.get(i).forEach(method -> {
+                String returnType = getReturnType.apply(method);
+                if (allStages.containsKey(returnType)) {
+                    allStages.get(returnType).getMethods().forEach(nextMethod -> {
+                        if (!used.contains(nextMethod)) {
+                            nextStage.add(nextMethod);
+                            used.add(nextMethod);
+                        }
+                    });
+                }
+            });
+            if (!nextStage.isEmpty()) {
+                results.add(nextStage);
+            }
+        }
+    }
+
+    @Override
+    public boolean isClassLevelChanged() {
+        return false;
+    }
+
+    @Override
+    protected void calcChangeLog() {
+        int oldSize = oldMethodStages.size() - 1;
+        int newSize = newMethodStages.size() - 1; // don't need check the last stage
+        for (int i = 0; i < Math.min(oldSize, newSize); ++i) {
+            for (JApiMethod method : oldMethodStages.get(i)) {
+                if (newMethodStages.get(i).contains(method)) {
+                    switch (method.getChangeStatus()) {
+                        case REMOVED:
+                            addClassTitle(breakingChange);
+                            breakingChange.add(String.format("* `%s` was removed in stage %d", MethodName.name(method.getOldMethod().get()), i + 1));
+                            break;
+                        case MODIFIED:
+                            if (!method.getOldMethod().get().getLongName().equals(method.getNewMethod().get().getLongName())) {
```

Comment:
Does `getLongName` return variable name (which is not counted as change)?

Reply:
Yes, `getLongName` return with function name with function parameter.